src/components/directory-tree-view.js

```diff
@@ -98,6 +98,12 @@ const DirectoryTreeView = ({ treeViewUuid, mapData, onContextMenu }) => {
             <TreeItem
                 key={node.elementUuid}
                 nodeId={node.elementUuid}
+                onIconClick={() => {
+                    toggleDirectory(node.elementUuid);
```

Comment:
add a handleIconClick function which :
call a prop function in `DirectoryTreeView` from `DirectoryTreeContainer` which update the `node.elementUuid` directory (ex: `onUpdateDirectory` ?) and then call  `toggleDirectory(node.elementUuid);`